include/gatekeeper_mailbox.h

```diff
@@ -19,14 +19,20 @@
 #ifndef _GATEKEEPER_MAILBOX_H_
 #define _GATEKEEPER_MAILBOX_H_
 
-/*
- * TODO Remove this preprocessing directive when there are
- * are calls to these functions; for now, the mailbox is
- * just an example of how to set up a library.
- */
-#if 0
-void create_mailbox(void);
-void destroy_mailbox(void);
-#endif
+#include <rte_ring.h>
+#include <rte_mempool.h>
+
+struct mailbox {
+	struct rte_ring    *ring;
+	struct rte_mempool *pool;
+};
+
+struct mailbox *create_mailbox(
+	const char *tag, int ele_count,
+	int ele_size, unsigned int lcore_id);
+void *mb_get_entry(struct mailbox *mb);
+int mb_put_entry(struct mailbox *mb, void *obj);
+void mb_release_entry(struct mailbox *mb, void *obj);
```

Comment:
Rename `mb_release_entry()` to `mb_free_entry()` in order to match `mb_alloc_entry()`.